zebedee-cms/src/main/java/com/github/onsdigital/zebedee/kafka/KafkaClient.java

```diff
@@ -0,0 +1,12 @@
+package com.github.onsdigital.zebedee.kafka;
+
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import java.util.concurrent.Future;
+
+/**
+ * Interface representing a client that sends messages to kafka
+ */
+public interface KafkaClient {
```

Comment:
Is this likely to ever have any other methods? If no can you add the 
```
@FunctionalInterface
```
Annotation please.

Reply:
It might have more methods in future if there are other producers needed. At the moment zebedee will only raise a `content-published` event but if in future it needed others (say a `user-logged-in` or something like that) then there would be more methods. I have perhaps muddied the water a little by using a functional lambda to define a no-op client when the feature is turned off, but my thought is that by that stage it would no longer need a feature flag. I could replace this with a no op class instead if it made it clearer?